src/runtime.rs

```diff
@@ -223,6 +225,16 @@ where
         }
     }
 
+    /// Calls the callback if the current revision is cancelled.
+    ///
+    /// Observing cancellation may lead to inconsistencies in database storage,
+    /// so the callback must panic.
+    pub fn if_current_revision_is_canceled(&self, cb: fn() -> !) {
```

Comment:
I'm not sure this method is really *needed* -- I had sort of expected to just do something like this on the client side

```
if self.is_current_revision_canceled() {
  panic!()
}
```

But I guess the point of view this is to avoid adding any reads etc?

I'd still (I think) like to keep the `is_current_revision_canceled` infrastructure, since it's really not much complexity and people might not want to `panic!`.

Reply:
Yeah, if we keep the current infra, `is_current_revision_canceled` will be enough. If we *remove* current infra, than this API will be better, because it prevents you from being accidentally non-determentistic.